### DISCUSSION TITLE
Can't run venv

### DISCUSSION THREAD

**Discussion by Lewdoz** [netbox-community/netbox#10028] [Q&A] [answered]
When I run:
'''source /opt/netbox/venv/bin/python3'''
I get a result o:
'''/bin/sh: 136: source: not found'''

I have followed the instructions on Netbox.dev and Vultr, which worked just fine on an Ubuntu kernel. I'm working on Debian 7 now. I have attempted all of the fixed on the forums, reinstalling pip3, installing missing dependencies and I set up gunicorn. Any help or troubleshooting assistance would be greatly appreciated.

**Comment by Lewdoz** [comment]
I did start by manually patching the upgrade.sh file:
https://github.com/netbox-community/netbox/discussions/5813

**Comment by mtinberg** [comment, marked as answer]
Maybe I'm misreading but wouldn't you want

$ source /opt/netbox/venv/bin/activate

the "source" shell command is a built-in that reads a file and evals it, so you can import a bunch of variables into your current shell environment, and it's how python virtualenvs work, by setting the various python library path environment config variables to point inside the venv directory structure and not the default system paths, overriding them.

is your /bin/sh one that doesn't have the "source" built-in command, it might have one called "." instead eg.

$ . /opt/netbox/venv/bin/activate

that's the older name for this functionality IIRC


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Lewdoz ***@***.***>
Sent: Tuesday, August 16, 2022 2:57 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Can't run venv (Discussion #10028)


When I run:
'''source /opt/netbox/venv/bin/python3'''
I get a result o:
'''/bin/sh: 136: source: not found'''

I have followed the instructions on Netbox.dev and Vultr, which worked just fine on an Ubuntu kernel. I'm working on Debian 7 now. I have attempted all of the fixed on the forums, reinstalling pip3, installing missing dependencies and I set up gunicorn. Any help or troubleshooting assistance would be greatly appreciated.

**Reply by Lewdoz** [reply]
I did try both commands and got the following:
![image](https://user-images.githubusercontent.com/111309657/184997655-5deffeef-d036-46c8-b4f6-a37b02567682.png)

I went through my bin and venv folder and everything looks the same as what I have installed on the Ubuntu machine. I followed the steps I wrote down previously and the only difference was Python being 3.9 and having to manually set the time for Kiev.

**Reply by markkuleinio** [reply]
The command is `source /opt/netbox/venv/bin/activate` (you had extra $ and . in your commands)

**Reply by Lewdoz** [reply]
I get the same issue:
$ source /opt/netbox/venv/bin/activate
/bin/sh: 1: source: not found
$

**Reply by markkuleinio** [reply]
https://stackoverflow.com/questions/13702425/source-command-not-found-in-sh-shell

**Reply by Lewdoz** [reply]
Thanks, this lead me to the right question to ask.
Debian required me to use . instead of source
. /opt/netbox/venv/bin/activate
